src/SokuMod/Connection.cpp

```diff
@@ -309,8 +318,23 @@ bool Connection::_handlePacket(const Lobbies::PacketGameStart &packet, size_t &s
 	if (size < sizeof(packet))
 		return false;
 	size -= sizeof(packet);
-	if (this->onConnectRequest)
-		this->onConnectRequest(packet.ip, packet.port, packet.spectator);
+	if (this->onConnectRequest){
+		const char * ip = packet.ip;
+		unsigned short port = packet.port;
+		char ipv6[sizeof(packet.ipv6)+1];
+		char ipv6MappedIpv4[16];
+		if (isIpv6Available() && packet.port6){
+			strncpy(ipv6, packet.ipv6, sizeof(packet.ipv6));
+			ipv6[sizeof(ipv6)-1] = '\0';
+			printf("p1 ipv6 is: %s\n", ipv6);
+			if (mapIpv6toIpv4(ipv6, ipv6MappedIpv4, sizeof(ipv6MappedIpv4))){
```

Comment:
How does that work? Can you convert any IPv6 to IPv4 somehow!? I assume it's just the IPv6 mod doing some magic behind the scene?

Reply:
yeah. it is ipv6 mod that does that.
the ipv6 mod hookes network functions and converts virtual ipv4 addresses to ipv6 ones

Reply:
> Can you convert any IPv6 to IPv4 somehow!?

Not "real" IPv4 address